EventKit/eventkit-cloud
# Add favorite product model

## Description of Improvement

Adds the UserFavoriteProduct model. Adds the logic to support a user favoriting a data provider. 

## How to test

Add a favorite in the admin page and select a product. Check the that the 'favorite' annotation turns to True on that dataprovider in the API. 

## Quality Assurance 

The following items have been updated:

:smile:
